From cc4cb200c68448b22f13de3b49de7c37d10f04a9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20Michaj=C5=82ow?= <kasper93@gmail.com>
Date: Tue, 29 Sep 2026 11:34:18 +0200
Subject: [PATCH] avformat/dashdec: free the old IO buffer when reopening a
 representation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

After a failed reopen it is still there without a demuxer context, so
close_demux_for_component() does not free it.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>
---
 libavformat/dashdec.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavformat/dashdec.c b/libavformat/dashdec.c
index 740ac6bc1c15..2ce6fe6c5601 100644
--- a/libavformat/dashdec.c
+++ b/libavformat/dashdec.c
@@ -1964,6 +1964,7 @@ static int reopen_demux_for_component(AVFormatContext *s, struct representation
         pls->ctx = NULL;
         goto fail;
     }
+    av_freep(&pls->pb.pub.buffer);
     ffio_init_context(&pls->pb, avio_ctx_buffer, INITIAL_BUFFER_SIZE, 0,
                       pls, read_data, NULL, c->is_live ? NULL : seek_data);
     pls->pb.pub.seekable = 0;
